core/src/entities/enrichments/vulnerabilities.rs

```diff
@@ -1,16 +1,24 @@
 use crate::entities::enrichments::cvss::Summary;
+use crate::entities::tasks::{Task, TaskRef};
+use crate::Error;
 use serde::{Deserialize, Serialize};
 use serde_with::skip_serializing_none;
 use std::fmt::{Display, Formatter};
 
-///
+/// Alias for the native CycloneDx Severity enum.
 pub type Severity = crate::entities::cyclonedx::Severity;
 
 /// Identified security issue for a [Package].
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 #[skip_serializing_none]
 pub struct Vulnerability {
+    /// Unique identifier for instance.
+    pub id: String,
```

Comment:
Option<T> all around so we can define defaults and use those when no value is present

Reply:
I know you already know this

Reply:
Making a note, but I think we should push this to another PR.  It would delay this dramatically and explode the PR size.